fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonUtils.xtend

```diff
@@ -259,33 +259,42 @@ class PythonUtils {
 		«ENDIF»
 		«ENDIF»
 		«IF !cmd.argout.type.voidType»    dtype_out=«cmd.argout.type.pythonTypeHL», 
-		«IF !cmd.argout.description.empty»    doc_out="«cmd.argout.description.oneLineString»", 
-		«ENDIF»
-		«ENDIF»
-		«IF !cmd.description.empty»    description="«cmd.description.oneLineString»",
 		«ENDIF»
     «setAttrPropertyHL("display_level", cmd.displayLevel, false)»
     «setAttrPropertyHL("polling_period", cmd.polledPeriod, false)»
 		«IF cmd.hasCommandArg»    )
 		«ENDIF»
+		«ENDIF»
 	«ENDIF»
-«ENDIF»
     @DebugIt()
     def «cmd.methodName»(self«IF !cmd.argin.type.voidType», argin«ENDIF»):
         «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
-            «IF !cmd.argout.type.voidType»
-                «protectedAreaHL(cls, cmd.name, cmd.argout.type.defaultValueReturnHL, false)»
-            «ELSE»
-                pass
-            «ENDIF»
+        «openProtectedAreaHL(cls,cmd.name)»
+        """
+        «cmd.description.commentCmdParamMultiLines»
```

Comment:
The docstring is inside the protected area, and usually that part of the template is empty.  What happens if you modify the docstring in the .py file, and then regenerate?

A benefit of putting it inside the protected area, is that modifications by the user would be preserved in the .py.  The downside is that the .xmi file won't have this updated info.

Reply:
Agree with your concern. It is a regeneration issue which I am not handling. I have added docstrings for commands in protected region for now so that changes remain in the .py file if regenerated.